### PR TITLE
migrated to webpack 2.

### PR DESCRIPTION
**Pull request by nickytonline**

### CODE REVIEW

**Comment by nickytonline** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://34c00e44-92fb-4d49-9c3b-fb947b195257.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/nickytonline/ts-preact-starter/branch/feature/webpack-2/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

**Comment by nickytonline** [comment]
Implements #1.

**Comment by nickytonline** [comment]
React Storybook's webpack config stays in 1.x format because the project hasn't upgraded to webpack 2 yet, https://github.com/storybooks/react-storybook/blob/master/package.json#L61

File: package.json [hunk 1]
```diff
@@ -41,8 +41,8 @@
     "tslint": "4.1.1",
     "tslint-loader": "3.3.0",
     "typescript": "2.1.4",
-    "webpack": "1.14.0",
-    "webpack-dev-server": "1.16.2"
+    "webpack": "2.1.0-beta.28",
```
Comment (Hotell) [inline package.json]: both are in RC, why using beta?

File: webpack.config.js [hunk 2]
```diff
@@ -1,65 +1,65 @@
...
-  tslint: {
-    configFile: './tslint.json',
-    emitErrors: true,
-    failOnHint: true
-  },
-  devServer: {
+  ]
+};
+
+if (environment === 'dev') {
```
Comment (Hotell) [inline webpack.config.js]: everytime you `if` in webpack config a kitten dies :D

consider https://github.com/kentcdodds/webpack-config-utils

Comment (Hotell) [inline webpack.config.js]: like this 
https://github.com/ngParty/React-scaffold/blob/develop/webpack.config.js#L53

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
   ],
   "scripts": {
     "build": "cross-env NODE_ENV=prod webpack",
-    "start": "cross-env NODE_ENV=dev webpack-dev-server --progress --debug --config --display-error-details",
+    "start": "cross-env NODE_ENV=dev webpack-dev-server --progress --debug --config ./webpack.config.js",
```
Comment (Hotell) [inline package.json]: cross-env bye bye https://github.com/ngParty/React-scaffold/blob/develop/package.json#L13

Comment (nickytonline) [inline package.json]: Thanks for the feedback. I didn't know about this. I'll go ahead and update this this week.

File: package.json (hunk 1 above)
Comment (nickytonline) [inline package.json]: @Hotekk, I was going by the migrate to webpack 2 docs. I'll upgrade to the RC versions.

File: webpack.config.js (hunk 2 above)
Comment (nickytonline) [inline webpack.config.js]: Didn't know about this package. Thanks!

File: package.json (hunk 1 above)
Comment (nickytonline) [inline package.json]: Looks like RC 3 just landed.

File: webpack.config.js (hunk 2 above)
Comment (Hotell) [inline webpack.config.js]: sure thing! good job mister!

**Pull request merged**